### ISSUE TITLE
Incorrect definition for window.onerror

### ISSUE THREAD

**Issue by niemyjski** [Microsoft/TypeScript#2838]
There is a new spec for window.onerror ( https://html.spec.whatwg.org/multipage/webappapis.html#errorevent ). I came across this when I tried to implement a custom handler in typescript and the definition was incorrect/outdated. I got this link from https://danlimerick.wordpress.com/2014/01/18/how-to-catch-javascript-errors-with-window-onerror-even-on-chrome-and-firefox/

**Comment by niemyjski** [comment]
If you want, I could do a pull request for this.

**Comment by niemyjski** [comment]
Do you think I should remove the existing definitions for this and update the interface to be this new spec?

**Comment by niemyjski** [comment]
I'm also wondering if the handler name should be renamed to what the spec is.

**Comment by niemyjski** [comment]
But this change would break people if I did either of the two and you were using a custom handler (not likely)

**Comment by mhegazy** [comment]
Can you check the latest [lib.d.ts](https://github.com/Microsoft/TypeScript/blob/master/bin/lib.d.ts#L15441) version. how is that diffrent from the documentation?

**Comment by niemyjski** [comment]
Hello,

The latest is:

```typescript
interface ErrorEventHandler {
    (event: Event, source?: string, fileno?: number, columnNumber?: number): void;
    (event: string, source?: string, fileno?: number, columnNumber?: number): void;
}
```
and it should be according to the docs ( https://html.spec.whatwg.org/multipage/webappapis.html#errorevent ) 

```typescript
interface ErrorEventHandler {
    (message: string, filename?: string, lineno?: number, colno?: number, error?:Error): void;
}
```
I'd prefer the names to be cleaned up a bit:

```typescript
interface ErrorEventHandler {
    (message: string, fileName?: string, line?: number, column?: number, error?:Error): void;
}
```

**Comment by mhegazy** [comment]
I see. thanks for the explanation. These typings are auto generated so we can not change these definitions manually. @zhanbos is working on a script to allow changing them. once this happens a PR would be welcomed.

**Comment by niemyjski** [comment]
Just let me know when you want me to change these and I'll do a pull request :)

**Comment by stimms** [comment]
I have also just encountered this issue. Is there anything we can do to help @zhanbos out and make sure this fix makes 1.6?
